public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+
+function FilterSelectorRow(props: any) {
+  const { fieldNameOptions, onDelete, onConfigChange, config, fieldType } = props;
+
+  console.log('props', props);
+
+  // Find filter types that fit the chosen field type
+  let filterTypeOptions = valueFiltersRegistry
+    .list()
+    .filter(element => {
+      if (element.supportedFieldTypes) {
+        return !element.supportedFieldTypes ? true : element.supportedFieldTypes.includes(fieldType);
+      } else {
+        return true;
+      }
+    })
+    .map(item => ({ value: item.id, label: item.name, description: item.description }));
+
+  let filterInfo = valueFiltersRegistry.get(config.filterType);
+  let filterValid = filterInfo.getInstance({
+    filterExpression: config.filterExpression,
+    fieldType: fieldType,
+  }).isValid;
+
+  let fieldNameInvalid =
+    config.fieldName === null ||
+    fieldNameOptions.filter((item: Record<string, any>) => item.value === config.fieldName).length === 0;
+  let filterTypeInvalid =
+    !fieldNameInvalid && filterInfo.supportedFieldTypes && !filterInfo.supportedFieldTypes.includes(fieldType);
+  let filterExpressionInvalid = !fieldNameInvalid && !filterTypeInvalid && !filterValid;
+
+  let filterOptionsInput = null;
+  if (filterInfo.placeholder) {
+    filterOptionsInput = (
+      <>
+        <Input
+          className="flex-grow-1"
+          invalid={filterExpressionInvalid}
+          defaultValue={config.filterExpression}
+          placeholder={filterInfo.placeholder}
+          onBlur={event => {
+            onConfigChange({ ...config, filterExpression: event.currentTarget.value });
+          }}
+        />
+      </>
+    );
+  }
+
+  return (
+    <div className="gf-form-inline">
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter Type</div>
+        <Select
+          className="width-8"
+          options={[
+            { label: 'Include', value: 'include' },
+            { label: 'Exclude', value: 'exclude' },
+          ]}
+          value={config.type}
+          onChange={option => {
+            // console.log('onChange filterType', option.value);
+            onConfigChange({ ...config, type: option.value });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter on Field</div>
+        <Select
+          className="width-16"
+          placeholder="Field Name"
+          options={fieldNameOptions}
+          value={config.fieldName}
+          invalid={fieldNameInvalid}
+          onChange={value => {
+            // console.log('onChange fieldName', value);
+            if (value === null) {
+              onConfigChange({ ...config, fieldName: null });
+            } else {
+              onConfigChange({ ...config, fieldName: value.value });
+            }
```

Comment:
```suggestion
           onConfigChange({
              ...config,
              fieldName: value?.value ?? null,
            })
```

Here is a smart trick to make the code a bit more compact. I am using the `value?.value` that is a safe way of accessing properties on an object when it might be undefined. If `value` if undefined it will return undefined instead of throwing an exception. I combine this together with the `??` that will return the value to the right if the value to the left is undefined so it will then be null.